### PR TITLE
Add support for JSX dev runtime

### PR DESCRIPTION
**Pull request by remcohaszing**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

This adds support for the JSX automatic development runtime.

### CODE REVIEW

File: jsx-dev-runtime.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+   *   (name: null|undefined, props: {children?: XChild}, ...unused: unknown[]): Root
+   *   (name: string, props: JSXProps, ...unused: unknown[]): Element
+   * }}
+   */
+  (
+    function (name, props) {
+      return jsx(/** @type {string} */ (name), props)
+    }
+  )
+export {Fragment} from './jsx-runtime.js'
```
Comment (wooorm) [inline jsx-dev-runtime.js]:
```suggestion

export {Fragment} from './jsx-runtime.js'
```

Feels a bit cleaner?

File: script/generate-jsx.js [hunk 2]
```diff
@@ -38,10 +38,26 @@ fs.writeFileSync(
...
+
+fs.writeFileSync(
+  path.join('test', 'jsx-build-jsx-automatic-development.js'),
+  toJs(
+    // @ts-expect-error it’s a program.
+    buildJsx(
+      // @ts-expect-error Acorn nodes are assignable to ESTree nodes.
+      Parser.extend(acornJsx()).parse(
+        doc.replace(/'name'/, "'jsx (estree-util-build-jsx, automatic)'"),
+        {sourceType: 'module', ecmaVersion: 2021}
```
Comment (wooorm) [inline script/generate-jsx.js:55]: This might error (it just did for me on the above ones, see the `ts-ignore` for more info

Comment (wooorm) [reply script/generate-jsx.js:55]: oh it passes, weird 🤷‍♂️ well nvm then

File: jsx-dev-runtime.js (hunk 1 above)
Comment (remcohaszing) [reply jsx-dev-runtime.js]: True. `npm test` autofixed it into this.

File: script/generate-jsx.js (hunk 2 above)
Comment (remcohaszing) [reply script/generate-jsx.js:55]: Yeah I had to resolve some merge conflicts here. Not sure what happened upstream.

**Review by ChristianMurphy — APPROVED** [review]

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thanks, released in [3.1.0](https://github.com/syntax-tree/xastscript/releases/tag/3.1.0)!

**Pull request merged**